htdocs/core/class/html.formaccounting.class.php

```diff
@@ -329,7 +330,7 @@ public function select_account($selectid, $htmlname = 'account', $showempty = 0,
 					}
 					// Remember guy's we store in database llx_facturedet the rowid of accounting_account and not the account_number
 					// Because same account_number can be share between different accounting_system and do have the same meaning
-					if ($selectid != '' && $selectid == $select_value_in) {
+					if ($selectid != "" && intval($selectid) == intval($select_value_in)) {
```

Comment:
obj->account_number  so selectid can be a string so it should be better to make
`if ($selectid != "" && ((string) $selectid == (string) $select_value_in))`